caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -120,28 +120,131 @@ def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
 
         return cls.bootstrap(model_name_or_path=artifacts_path)
 
+    def _truncate_input_tokens(
+        self, truncate_input_tokens, texts: List[str]
+    ) -> List[str]:
+        """Truncate input tokens
+        Args:
+            truncate_input_tokens: int
+                Truncation length for input tokens.
+                If less than zero, this is disabled (returns texts without processing).
+                If zero or greater than the model's maximum, then this is a test
+                to see if truncation is needed. If needed, an exception is thrown.
+                Otherwise, we take this usable truncation limit to truncate the tokens and then
+                decode them to return truncated strings that can be used with this model.
+            texts: List[str]
+                Input texts to be checked and optionally truncated.
+        Returns:
+            List[str]: the texts after checking and/or truncating
+        """
+
+        # NOTE: When inference is called immediately after load (typical case with lazy loading),
+        # using the tokenizer right away here results in a warning like:
+        #     huggingface/tokenizers: The current process just got forked, after parallelism
+        #     has already been used. Disabling parallelism to avoid deadlocks...
+        #     To disable this warning, you can either:
+        #     - Avoid using `tokenizers` before the fork if possible
+        #     - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
+        # A warmup encode() call in load() will take care of this (the first option above).
+        # This here comment is in case we need to set TOKENIZERS_PARALLELISM in the future.
+
+        if truncate_input_tokens < 0:
+            return texts
+
+        max_tokens = self.model.max_seq_length
+
+        # Do truncation if given a usable truncation value, else test for need to truncation
+        if 0 < truncate_input_tokens <= max_tokens:
+            okay_to_truncate = True
+            max_length = truncate_input_tokens
+            ret = []  # will return truncated texts
+        else:
+            okay_to_truncate = False
+            max_length = max_tokens
+            ret = texts  # will not alter texts when truncation is not allowed
+
+        for text in texts:
+            tokenized = self.model.tokenizer(
+                text,
+                return_attention_mask=False,
+                return_token_type_ids=False,
+                return_overflowing_tokens=True,
```

Comment:
I don't actually use the tokens, but this setting gives me the overflow which indicates that there were extra tokens. This allows me to (by default) detect overflow, count up the total, and raise the error.  This is the primary goal... to make sentence-transformers fail like other models instead of silently truncating when the model max is exceeded.  The error message is expected to provide the token count because it isn't exactly character count.  

Since I also allow a truncation setting <= model_max (when the client sends that parameter). This setting also gives me the input_ids that I'll use for that allowed truncation.

The best alternative I considered was to pad the max and then do the truncation myself, but I think the current way provides a better way to truncate and a better way to detect overflow and report the total length.